src/UserPage/index.tsx

```diff
@@ -1,39 +1,237 @@
-import { useCallback } from 'react';
-import { Trans, useTranslation } from 'react-i18next';
+import { useCallback, useState } from 'react';
+import { TFuncKey, Trans, useTranslation } from 'react-i18next';
+import { useParams } from 'react-router';
+import Button, { ButtonContainer } from '../Form/Button';
 import { Checkbox } from '../Form/Checkbox';
+import { Input, Textarea } from '../Form/Input';
+import InputContainer from '../Form/InputContainer';
 import Header from '../Header';
 import { useUser, useUserSettings } from '../hooks/appContext';
-import { collections } from '../hooks/data';
-import { ColumnHeading, SingleColumn } from '../Layout/Columns';
-import { UserSettings } from '../types';
+import { collections, useDocument } from '../hooks/data';
+import { ColumnHeading, Columns } from '../Layout/Columns';
+import LinkButton from '../LinkButton';
+import { ModalDialog } from '../ModalDialog';
+import { User, UserSettings } from '../types';
+import { removeHttpsWwwPageParams } from '../utils';
+import s from './style.module.css';
+
+const USER_LINKS: {
+    type: 'twitter' | 'website' | 'instagram';
+    getUrl: (handle?: string) => string;
+    getLinkLabel: (handle?: string) => string;
+    labelKey: TFuncKey<'translation'>;
+    placeholerKey: TFuncKey<'translation'>;
+    labelHintKey: TFuncKey<'translation'>;
+}[] = [
+    {
+        type: 'instagram',
+        getUrl: (handle?: string) => `https://www.instagram.com/${handle}`,
+        getLinkLabel: (handle?: string) => `@${handle}`,
+        labelKey: 'userPage.socialLabels.instagram.label',
+        placeholerKey: 'userPage.socialLabels.instagram.placeholder',
+        labelHintKey: 'userPage.socialLabels.instagram.inputHint',
+    },
+    {
+        type: 'twitter',
+        getUrl: (handle?: string) => `https://twitter.com/${handle}`,
+        getLinkLabel: (handle?: string) => `@${handle}`,
+        labelKey: 'userPage.socialLabels.twitter.label',
+        placeholerKey: 'userPage.socialLabels.twitter.placeholder',
+        labelHintKey: 'userPage.socialLabels.twitter.inputHint',
+    },
+    {
+        type: 'website',
+        getUrl: (handle?: string) => `${handle}`,
```

Comment:
I think this needs some kind of validation. Right now it's very easy to produce broken relative-links.